crates/ironrdp-acceptor/src/credssp.rs

```diff
@@ -64,6 +68,30 @@ impl CredentialsProxy for CredentialsProxyImpl<'_> {
     }
 }
 
+pub(crate) async fn resolve_generator(
+    generator: &mut CredsspProcessGenerator<'_>,
+    network_client: &mut dyn AsyncNetworkClient,
+) -> Result<ServerState, ServerError> {
+    let mut state = generator.start();
+
+    loop {
+        match state {
+            GeneratorState::Suspended(request) => {
+                let response = network_client
+                    .send(&request)
+                    .await
+                    .inspect_err(|err| error!(?err, "Failed to send a Kerberos message"))
```

Comment:
question: Do we really need `inspect_err` here? I think we are propagating the error using `map_err`.

Reply:
> I think we are propagating the error using map_err.

Yes, but `map_err` uses the `Display` error implementation, but in the `error!` macro uses error's `Debug` implementation. It can give us more details about the error.

Reply:
got it. removed \`inspect_err\`